hoodie-client/src/main/java/com/uber/hoodie/io/HoodieCreateHandle.java

```diff
@@ -80,6 +82,7 @@ public HoodieCreateHandle(HoodieWriteConfig config, String commitTime, HoodieTab
       String partitionPath, String fileId, Iterator<HoodieRecord<T>> recordIterator) {
     this(config, commitTime, hoodieTable, partitionPath, fileId);
     this.recordIterator = recordIterator;
+    this.useWriterSchema = true;
```

Comment:
why do we have this flag?

Reply:
This flag is required because we use the same code path for compaction as well. The records passed in from a compaction action already have the hoodie commit metadata fields, so in that case we need to force use writerSchema.